flit/upload.py

```diff
@@ -27,6 +31,15 @@
     "http://upload.pypi.io/",
 )
 
+
+@dataclass
+class RepoDetails:
+    url: str
+    username: Optional[str] = None
+    password: Optional[str] = None
+    is_warehouse: bool = True
```

Comment:
Do you want to update that naming to `is_legacy`?

Reply:
`is_pypi` might be a better name now - it's distinguishing PyPI (the new site, aka Warehouse) from other servers like devpi, and originally from PyPI classic.

It's only used for the 'Package is at' log message, so it's not too important that it's always right.